daemon/src/model/cfd.rs

```diff
@@ -1276,6 +1279,273 @@ impl Cfd {
+                },
+                Command::ConfirmLock => vec![Event::LockConfirmed],
+                Command::ConfirmCommit => vec![Event::CommitConfirmed],
+                Command::ConfirmCet => vec![Event::CetConfirmed],
+                Command::ConfirmRefund => vec![Event::RefundConfirmed],
+                Command::ConfirmCollaborativeSettlement => {
+                    vec![Event::CollaborativeSettlementConfirmed]
+                }
+                Command::HandleAttestation(attestation) => {
+                    // TODO: The way I imagine this:
+                    //  Rather than having function `cet(&self) -> Option<Cet>` on the aggregate we
+                    // explicitly transition into having a ready cet
+                    //  I find "tell don't ask" cleaner here, i.e. event that tells the cet is ready
+                    // rather than check from the outside if "a cet is available".
+                    //  Note: When handling CetReadyForPublication on the outside we would
+                    // apply+save the event and then publish the tx.
+                    if self.cet_finality {
+                        vec![
+                            Event::OracleAttested(attestation),
+                            Event::CetReadyForPublication(todo!("pick CET")),
+                        ]
+                    } else {
+                        vec![Event::OracleAttested(attestation)]
+                    }
+                }
+                Command::ConfirmCetTimelock => {
+                    if self.attestation.is_some() {
+                        vec![
+                            Event::CetTimelockConfirmed,
+                            Event::CetReadyForPublication(todo!("pick CET")),
+                        ]
+                    } else {
+                        vec![Event::CetTimelockConfirmed]
+                    }
+                }
+                Command::ConfirmRefundTimelock => todo!(),
+            }
+        }
+    }
+
+    pub enum Command {
+        CompleteContractSetup(Result<Dlc, ContractSetupError>),
+
+        // TODO: Should this `Complete` represent Accepted/Rejected/Failed, or should we model
+        // accept/reject separated from setup exchange?  Error could represent rejected as
+        // well
+        CompleteRollover(Result<(Dlc, RevokedCommit)>),
+
+        // TODO: Should this `Complete` represent Accepted/Rejected/Failed, or should we model
+        // accept/reject separated from setup exchange?  Error could represent rejected as
+        // well
+        CompleteCollaborativeSettlement(Result<CollaborativeSettlement>),
+
+        // monitoring
+        ConfirmLock,
+        ConfirmCommit,
+        ConfirmCet,
+        ConfirmRefund,
+        ConfirmCollaborativeSettlement,
+
+        ConfirmCetTimelock,
+        ConfirmRefundTimelock,
+
+        // oracle
+        HandleAttestation(Attestation),
+    }
+
+    pub enum Event {
+        ContractSetupCompleted(Dlc),
```

Comment:
Each variant needs to have the CFDs ID + a timestamp.

Reply:
I can model it to be more driven by the `Event`.

My thinking so far was:
When we handle a command we already know for which cfd it was emitted. 
The `Event` is emitted by the `Aggregate`. The `Aggregate` knows it's own id, so why would the `Event` need it?
But that will be annoying when saving things (...). Probably better to model it on the `Event` and make the `Event` the citizen that is concerned with this.

Reply:
We need to store the ID with the event so the aggregate that emits it needs to embed it otherwise we lose the association.